security/fma-generic-contracts.md

```diff
@@ -0,0 +1,147 @@
+
+- **Risk Assessment:**
+  High severity due to arbitrary slot bounds.
+  Low likelihood since this is well looked out for.
+
+- **Mitigations:**
+  Storage layouts are validated to only extend and not rewrite the slots of the previous version.
+
+- **Detection:**
+  If this error were missed, it could be detected by looking at the storage values in the proxy or recognizing an inconsistency in variables stored in overwritten slots.
+  Alternatively, we could detect it from monitoring or user reports that something seems off/incorrect.
+
+- **Recovery Path(s):**
+  Deploy new implementation with a fixed storage layout and write new values to storage.
+  Requires an upgrade.
+
+
+#### Proxy Initialized to Wrong Values
+
+- **Description:**
+    The proxy is initialized with the wrong values.
+
+- **Risk Assessment:**
+  High severity since arbitrary values can be incorrectly used.
+  Low likelihood since proxy initialization has a small surface area and is well verified/tested.
+
+- **Mitigations:**
+  Validate initialized values through extensive unit testing.
+  Simulations prior to performing the upgrade, verifying the initialized values.
+
+- **Detection:**
+  A simulation can show an unintended initialized proxy value.
+  Monitoring or user reports.
+
+- **Recovery Path(s):**
+  Repair the upgrade path (if needed).
+  Upgrade the smart contract.
+
+
+#### Can’t Upgrade Implementation
+
+- **Description:**
+  Contracts are unable to be upgraded.
+
+- **Risk Assessment:**
+  High severity since it effectively bricks an arbitrary contract in the system.
+  Low likelihood since contract upgrades are verified and follow industry-wide, standardized patterns.
+
+- **Mitigations:**
+  Validate through extensive unit testing.
+  Perform simulations that demonstrate upgradability.
+
+- **Detection:**
+  Simulate a contract upgrade to check for failure.
+
+- **Recovery Path(s):**
+  Upgrade contracts and/or proxies.
+  Update `op-chain-ops` if needed.
+
+
+#### Able to Reinitialize Implementation
+
+- **Description:**
+  A third party is able to reinitialize the contract to arbitrary values.
+
+- **Risk Assessment:**
+  High severity since an arbitrary actor can perform an action with arbitrary values.
+  Low likelihood since authentication is a critical, tested and validated property of the system.
+
+- **Mitigations:**
+  Validate through extensive unit testing.
+  Fuzz testing with random values.
+  Invariant testing that authentication properties hold.
+
+- **Detection:**
+  Simulate an arbitrary reinitialization.
+
+- **Recovery Path(s):**
+  Perform an upgrade that prevents the reinitialization.
+
+
+#### Backwards-incompatible ABIs
+
+- **Description:**
+  Functionality expected to hold across an upgrade or update breaks compatibility with prior systems.
+
+- **Risk Assessment:**
+  Low severity since forward-facing systems replace backwards systems.
```

Comment:
I'd say the actual reason this is low severity because it's a liveness failure, and not a safety failure